src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -0,0 +1,1235 @@
+            /// <remarks>
+            /// The buffer is only retained for a short time. If it has already been cleared, the result of this property is an empty sequence.
+            /// </remarks>
+            ReadOnlySequence<byte> OriginalMessagePack { get; }
+        }
+
+        /// <inheritdoc/>
+        JsonRpc IJsonRpcInstanceContainer.Rpc
+        {
+            set
+            {
+                Verify.Operation(this.rpc == null, "This formatter already belongs to another JsonRpc instance. Create a new instance of this formatter for each new JsonRpc instance.");
+
+                this.rpc = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="MultiplexingStream"/> that may be used to establish out of band communication (e.g. marshal <see cref="IDuplexPipe"/> arguments).
+        /// </summary>
+        public MultiplexingStream? MultiplexingStream
+        {
+            get => this.duplexPipeTracker.MultiplexingStream;
+            set
+            {
+                Verify.Operation(this.rpc == null, Resources.FormatterConfigurationLockedAfterJsonRpcAssigned);
+                this.duplexPipeTracker.MultiplexingStream = value;
+            }
+        }
+
+        /// <summary>
+        /// Sets the <see cref="MessagePackSerializerOptions"/> to use for serialization of user data.
+        /// </summary>
+        /// <param name="options">The options to use. Before this call, the options used come from <see cref="StandardResolverAllowPrivate.Options"/>.</param>
+        public void SetMessagePackSerializerOptions(MessagePackSerializerOptions options)
+        {
+            Requires.NotNull(options, nameof(options));
+
+            (this.userDataSerializationOptions, this.dynamicObjectTypeFormatterForUserSuppliedResolver) = this.MassageUserDataOptions(options);
+        }
+
+        /// <inheritdoc/>
+        public JsonRpcMessage Deserialize(ReadOnlySequence<byte> contentBuffer) => MessagePackSerializer.Deserialize<JsonRpcMessage>(contentBuffer, this.messageSerializationOptions);
+
+        /// <inheritdoc/>
+        public void Serialize(IBufferWriter<byte> contentBuffer, JsonRpcMessage message)
+        {
+            if (message is IJsonRpcMessageWithId msgWithId && (message is Protocol.JsonRpcResult || message is Protocol.JsonRpcError))
+            {
+                this.duplexPipeTracker.OnResponseSent(msgWithId.RequestId, successful: msgWithId is Protocol.JsonRpcResult);
+            }
+
+            if (message is Protocol.JsonRpcRequest request && request.Arguments != null && request.ArgumentsList == null && !(request.Arguments is IReadOnlyDictionary<string, object?>))
+            {
+                // This request contains named arguments, but not using a standard dictionary. Convert it to a dictionary so that
+                // the parameters can be matched to the method we're invoking.
+                request.Arguments = GetParamsObjectDictionary(request.Arguments);
```

Comment:
In the TestWindow, we have always used `new object[]{p0,p1}` for our arguments. It that using named arguments would be both larger and much slower. Is that your experience, and should we note that in the README?

Reply:
Named arguments do require some additional machinery, it's true. But on the whole of an RPC communication I'm not aware of significant perf issues that are unique to named args. It's something we can measure.
FWIW I find positional args generally more familiar than named args, as virtually every language uses positional args rather than named args by default. But we support named args because some servers _require_ them (e.g. LSP). So rather than document a recommendation, I think the best thing we can do is support both as well as possible and let folks use what they need to.
Also in light of #288 named args might become more preferred as they may allow for greater version compatibility across method signature changes, just FYI.